pinecone/client.go

```diff
@@ -1343,6 +1343,125 @@ func (i *InferenceService) Embed(ctx context.Context, in *EmbedRequest) (*infere
 	return decodeEmbeddingsList(res.Body)
 }
 
+// Document is a map representing the document to be reranked.
+type Document map[string]string
+
+// RerankRequest holds the parameters for calling [InferenceService.Rerank] and reranking documents
+// by a specified query and model.
+//
+// Fields:
+//   - Model: "The [model] to use for reranking.
+//   - Query: (Required) The query to compare with documents.
+//   - Documents: (Required) A list of Document objects to be reranked.
+//   - RankFields: (Optional) A list of document fields to use for ranking.
```

Comment:
This is SUCH a nit, but I think it might make it even _more_ clear if you said "A list of fields in each Document" instead of "A list of document fields"